### PR TITLE
Diataxis Reorganization

### PR DESCRIPTION
**Pull request by crutchcorn**
# Quick Preview

image

## TODO

- [x] Make sidebar conditional based on which subroute
- [x] Redirects for existing content
- [x] Redirects for subpaths
- [x] Fix links

### CODE REVIEW

**Comment by jasonLaster** [comment]
Lets get this new structure out there

**Comment by crutchcorn** [comment]
@jasonLaster I didn't bring back the design principles page, it was there before this PR and I just didn't remove it in this one yet. We can probably safely remove it.

**Review by jonbell-lot23 — COMMENTED** [review]
I wonder if a version of this page:
https://replay-documentation-git-diataxis-tracking-replayio.vercel.app/reference/test-runners/overview

Can be in Basics / Test Suite as the top page? It feels strange to have a DevTools/Overview but no Test Suites/Overview. 

This would partially address https://linear.app/replay/issue/DES-1346/testsuites-needs-better-quickstart-docs

**Comment by filiphric** [comment]
found a small glitch on mobile

IMG_8385

**Comment by filiphric** [comment]
Comparisons pages are also broken for some reason
Screenshot 2024-05-27 at 12 38 08

**Comment by filiphric** [comment]
There’s also some weir inception going on at `/reference/test-runners/playwright/examples` page

Screenshot 2024-05-27 at 12 39 10

**Comment by bvaughn** [comment]
On the Learn page, all of the links under "Comparison" are broken

https://www.loom.com/share/d086889c3ac44a80bff3f627f22c98e3

File: src/app/basics/test-suites/recent-runs/page.md [hunk 1]
```diff
@@ -92,14 +92,14 @@ Clicking on a recording opens the replay so that you can inspect it with Replay
...
   description="The Replay GitHub PR comment makes it easy to view a PR's recent runs."
 /%}
 
 {% quick-link
   title="Top failing and flaky tests"
   icon="sort"
-  href="/test-suites/top-failing-and-flaky-tests"
+  href="/basics/test-suites/top-failing-and-flaky-tests"
   description="Go to the Failing and Flaky Tests view to see your top failing and flaky tests."
 /%}
```
Comment (bvaughn) [inline src/app/basics/test-suites/recent-runs/page.md:104]: Totally don't need to fix this as part of this PR but while I'm noticing it, the screenshots for Recent Runs is outdated

| Shown | Actual |
| :--- | :--- |
| image | Screenshot 2024-05-27 at 7 19 11 AM |

cc @filiphric

Comment (filiphric) [reply src/app/basics/test-suites/recent-runs/page.md:104]: noted [DEV-534](https://linear.app/replay/issue/DEV-534/update-recent-runs-image), thanks 👍

File: src/app/basics/getting-started/record-your-cypress-test/page.md [hunk 2]
```diff
@@ -1,6 +1,5 @@
 ---
-title: Record your first replay
-description: Recording your test with the Replay browser is as simple as installing a Cypress plugin. This doc shows how you can record your first test in under two minutes.
+title: Record your Cypress.io test
```
Comment (bvaughn) [inline src/app/basics/getting-started/record-your-cypress-test/page.md:2]:
```suggestion
title: Record your Cypress.io tests
```

File: src/app/basics/getting-started/record-your-playwright-test/page.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Playwright
+title: Record your Playwright test
```
Comment (bvaughn) [inline src/app/basics/getting-started/record-your-playwright-test/page.md:2]:
```suggestion
title: Record your Playwright tests
```

**Comment by jasonLaster** [comment]
This is probably controversial because the content could be better at introducting Replay and discussing the differences between runtime and session replay, but either way, we should start with the introduction before getting started

Screenshot 2024-05-27 at 7 53 30 AM
fa5b71-4286-40f4-a444-63392b08c8bf">

**Comment by jonbell-lot23** [comment]
> Not opposed @jonbell-lot23, but might be room for a second PR to do so?
> 
> Only requesting, as this PR is pretty big and worried about inevitable future conflicts

Yup yup, let's do a follow-on PR

**Comment by crutchcorn** [comment]
The Broken Links checker is bugging atm

**Comment by jasonLaster** [comment]
Looks like this link is flaky:
https://developer.chrome.com/docs/devtools/recorder/

Should be safe to land after a rebase

File: src/app/basics/exploration/why-time-travel/page.md [hunk 4]
```diff
@@ -9,36 +9,36 @@ Problems in complex web applications often depend on particular network, timing,
 
 Replay is the first deterministic browser, so all bugs recorded with Replay are **reproducible**, no matter how complex or rare they are. This means your engineers **don’t have to spend days** trying to reproduce a flaky test or rare customer support issue.
 
-[Learn more](/replay-devtools/overview)
```
Comment (jasonLaster) [inline src/app/basics/exploration/why-time-travel/page.md:12]: I'd change exploration to introduction. This is important for redirects so lets get this in first.

**Comment by crutchcorn** [comment]
Feedback from @jasonLaster: Expand all first level expandable headings

**Comment by crutchcorn** [comment]
Feedback from @jasonLaster: Un-hyphenate "time-travel"

**Review by jonbell-lot23 — APPROVED** [review]
This reorganisation is an obvious step forward, and I'm excited to get it live!

Other topics are being discussed elsewhere and shouldn't prevent this PR. (We need a testsuites overview page, and bvaughn has a suggestion for trimming back the Playwright docs.) Ship it!

**Pull request merged**